testonly/hammer/hammer.go

```diff
@@ -336,6 +340,11 @@ func (s *hammerState) updateContents(rev int64, leaves []*trillian.MapLeaf) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	// Sanity check on rev being +ve.
+	if rev < 1 {
+		panic(fmt.Sprintf("internal error: trying to update hammer state with contents for rev %d", rev))
```

Comment:
These two new checks are a good idea, but they probably indicate an error in the mapserver rather than an internal error in the test tool.  So I think they should emit an `error` that gets propagated out through `setLeaves` instead.

Reply:
Is there much value in continuing to hammer something that we've just discovered a bug in?
Propogating an error outwards will terminate the hammer in any case.

My goal here is to establish some invariants on the content we store.  I don't believe we should ever associate content with a negative revision number so am making the hammer stop if that seems to be the case.

Reply:
As you say, we're not going to carry on hammering because an error will stop the test.

In protecting the invariants on the local contents, you've actually added extra checks on the mapserver's behaviour (namely: that it never returns a zero/negative revision, and that it never returns a revision that goes backwards) -- which is great.

But all the other checks on the mapserver's behaviour result in an `error` being emitted from the relevant entrypoint test function; these ones should too.

Reply:
OK, consistency wins. :)